dockers-apt-repo.sh

```diff
@@ -1,17 +1,40 @@
-# this is just notes going to automate!
-# Add Docker's official GPG key:
-sudo apt-get update
-sudo apt-get install ca-certificates curl
-sudo install -m 0755 -d /etc/apt/keyrings
-sudo curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
-sudo chmod a+r /etc/apt/keyrings/docker.asc
 
-# Add the repository to Apt sources:
+# Creating encryption certificates
+if (  which ca-certificates curl )
+then
+  echo " ca-certificates curl already exist "
+else 
+  echo " Creating ca-certificates "
+  sudo apt-get update
+  sudo apt-get install ca-certificates curl
+fi
+
+# Adding and configure Docker's official GPG key
+if (  which gpg )
+then
+  echo " gpg key already exist "
+else 
+  echo " Installing the keyring program "
+  sudo install -m 0755 -d /etc/apt/keyrings 
+  echo " Configuring gpg keyring "
+  sudo curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
+  echo " Modifying gpg file permissions "
+  sudo chmod a+r /etc/apt/keyrings/docker.asc
```

Comment:
Can you explain what this conditional is doing.

Reply:
Installing, configuring, and setting permissions for gpg keyring.

Reply:
So you check if gpg is installed and if not do all this stuff. What happens if gpg is installed and the keyring fails?